instrumentation/openfeign-9.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/openfeign/OpenFeignAttributesGetter.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.openfeign;
+
+import feign.Request;
+import feign.Response;
+import io.opentelemetry.instrumentation.api.instrumenter.http.HttpClientAttributesGetter;
+import io.opentelemetry.instrumentation.api.instrumenter.net.NetClientAttributesGetter;
+import io.opentelemetry.instrumentation.api.instrumenter.rpc.RpcAttributesGetter;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+
+enum OpenFeignAttributesGetter
+    implements
+        RpcAttributesGetter<ExecuteAndDecodeRequest>,
+        NetClientAttributesGetter<ExecuteAndDecodeRequest, feign.Response>,
+        HttpClientAttributesGetter<ExecuteAndDecodeRequest, feign.Response> {
+  INSTANCE;
+
+  @Override
+  public String system(ExecuteAndDecodeRequest request) {
+    return "openfeign";
+  }
+
+  @Override
+  public String service(ExecuteAndDecodeRequest request) {
+    return request.getTarget().name();
+  }
+
+  @Override
+  public String method(ExecuteAndDecodeRequest request) {
```

Comment:
both rpc client and http client semantic conventions define `method` but give a completely different meaning to it, thus it is not possible to implement both of these in the same class

Reply:
I have separated this attributes getter as "RPC Getter" and "HTTP Client Getter".
Obviously, there is no "rpc.method" attribute in "RPC Getter", because openfeign is a declarative http client proxy,"http.method" attributes will be set as "POST","GET"... in "HTTP Client Getter".

Reply:
> both rpc client and http client semantic conventions define `method` but give a completely different meaning to it, thus it is not possible to implement both of these in the same class

I have separated this attributes getter as "RPC Getter" and "HTTP Client Getter".
Obviously, there is no "rpc.method" attribute in "RPC Getter", because openfeign is a declarative http client proxy,"http.method" attributes will be set as "POST","GET"... in "HTTP Client Getter".